runtime/src/main/java/org/corfudb/protocols/wireprotocol/ReadRequest.java

```diff
@@ -16,24 +12,20 @@
 @AllArgsConstructor
 public class ReadRequest implements ICorfuPayload<ReadRequest> {
 
-    final Range<Long> range;
+    final long address;
 
     /**
      * Deserialization Constructor from ByteBuf to ReadRequest.
      *
      * @param buf The buffer to deserialize
      */
     public ReadRequest(ByteBuf buf) {
-        range = ICorfuPayload.rangeFromBuffer(buf, Long.class);
-    }
-
-    public ReadRequest(Long address) {
-        range = Range.singleton(address);
+        address = ICorfuPayload.fromBuffer(buf, Long.class);
```

Comment:
fyi `ICorfuPayload.fromBuffer` can block. You can just serialize the message directly. 
Example:
address = buf.readLong();
buf.writeLong(address)

Reply:
Done.